### PR TITLE
Fix/javadoc and tests

### PR DESCRIPTION
**Pull request by eloigrndl**
- Javadoc on a major part of the project
 - Several tests added that were previously missing
 - Lot of refactoring (imports, variable names, etc)

### CODE REVIEW

File: app/src/main/java/com/ncnf/utilities/Helpers.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.ncnf.utilities;
+
+import java.util.concurrent.CompletableFuture;
+
+public class Helpers {
+
+    /**
+     * Method to combine the result of multiple CompletableFuture
+     */
+    public static CompletableFuture<Boolean> combine(CompletableFuture<Boolean> u1, CompletableFuture<Boolean> u2) {
```
Comment (HyPn0TiiK) [inline app/src/main/java/com/ncnf/utilities/Helpers.java:10]: That might need to be in a specific db class, either rename the class it to make its purpose clearer, or move it to the db related package

File: app/src/main/java/com/ncnf/utilities/event/EventRelevanceCalculator.java [hunk 2]
```diff
@@ -15,15 +15,23 @@
...
+    /**
+     * Public constructor for the calculator
+     * @param l initial event list
+     */
     public EventRelevanceCalculator(List<Event> l) {
 
         for (Event e : l) {
             if (e != null)
-                this.events.add((Event) e);
+                this.events.add(e);
```
Comment (HyPn0TiiK) [inline app/src/main/java/com/ncnf/utilities/event/EventRelevanceCalculator.java:26]: nice

File: app/src/main/java/com/ncnf/utilities/map/MapHandler.java [hunk 3]
```diff
@@ -122,6 +122,10 @@ public void update_markers(){
         clusterManager.cluster();
     }
 
+    /**
+     * Add the marker on the map for each event in the list (either its own marker or marker for a cluster of event)
+     * @param events List of events we want to add an marker
+     */
     private void addEventMarkers(List<Event> events){
```
Comment (HyPn0TiiK) [inline app/src/main/java/com/ncnf/utilities/map/MapHandler.java:129]: was it necessary to add doc to a private method?

File: app/src/main/java/com/ncnf/utilities/user/LocationService.java [hunk 4]
```diff
@@ -118,7 +118,7 @@ public void onLocationResult(LocationResult locationResult) {
                                     if (user1 != null) {
                                         GeoPoint geoPoint = new GeoPoint(location.getLatitude(), location.getLongitude());
                                         saveUserLocation(geoPoint, user1.getUuid());
-                                        user1.setLoc(geoPoint);
+                                        user1.setLocation(geoPoint);
```
Comment (HyPn0TiiK) [inline app/src/main/java/com/ncnf/utilities/user/LocationService.java:121]: good

File: app/src/main/java/com/ncnf/views/fragments/organization/OrganizationTabFragment.java [hunk 5]
```diff
@@ -190,7 +190,7 @@ private boolean verifyTextInput(EditText input) {
         return true;
     }
 
-    private void displayPopUp(View view, String errorText) {
+    private void displayPopUp(String errorText) {
```
Comment (HyPn0TiiK) [inline app/src/main/java/com/ncnf/views/fragments/organization/OrganizationTabFragment.java:193]: thanks but this will be changed

File: .idea/compiler.xml [hunk 6]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
   <component name="CompilerConfiguration">
-    <bytecodeTargetLevel target="1.8" />
+    <bytecodeTargetLevel target="11" />
   </component>
 </project>
```
Comment (emmadeno) [inline .idea/compiler.xml:6]: Pretty sure these should be in the gitignore, I don't know why they're suddenly showing up in our PR's...

File: app/src/test/java/com/ncnf/database/builders/DatabaseEventBuilderTest.java [hunk 7]
```diff
@@ -41,10 +40,10 @@
 public class DatabaseEventBuilderTest {
 
     private DatabaseEventBuilder eventBuilder = new DatabaseEventBuilder();
-    private DatabaseGroupBuilder groupbuilder = new DatabaseGroupBuilder();
+    private DatabaseGroupBuilder groupBuilder = new DatabaseGroupBuilder();
```
Comment (HyPn0TiiK) [inline app/src/test/java/com/ncnf/database/builders/DatabaseEventBuilderTest.java:43]: the devil's in the details

File: app/src/test/java/com/ncnf/database/builders/DatabaseEventBuilderTest.java [hunk 8]
```diff
@@ -41,10 +40,10 @@
...
 
-    Map<String, Object> publicEvent;
-    Map<String, Object> privateEvent;
+    Map<String, Object> event;
+    Map<String, Object> group;
```
Comment (HyPn0TiiK) [inline app/src/test/java/com/ncnf/database/builders/DatabaseEventBuilderTest.java:46]: feels good

File: .idea/misc.xml [hunk 9]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectRootManager">
+    <output url="file://$PROJECT_DIR$/build/classes" />
+  </component>
+</project>
```
Comment (emmadeno) [inline .idea/misc.xml:6]: same here

File: .idea/runConfigurations.xml [hunk 10]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="RunConfigurationProducerService">
+    <option name="ignoredProducers">
+      <set>
+        <option value="com.android.tools.idea.compose.preview.runconfiguration.ComposePreviewRunConfigurationProducer" />
+      </set>
+    </option>
+  </component>
+</project>
```
Comment (emmadeno) [inline .idea/runConfigurations.xml:10]: and here

**Review by HyPn0TiiK — APPROVED** [review]
Thanks, I'm glad I didn't have to do it!

File: app/src/main/java/com/ncnf/repositories/OrganizationRepository.java [hunk 11]
```diff
@@ -37,46 +36,19 @@ public OrganizationRepository(FirebaseDatabase db)  {
...
-     * @return A CompletableFuture wrapping a list containing the loaded Organization objects
-     */
-    public CompletableFuture<List<Organization>> loadMultipleOrganizations(List<String> uuidList){
-        return this.db.whereIn(ORGANIZATIONS_COLLECTION_KEY, UUID_KEY, uuidList, Organization.class);
-    }
-
-    /**
-     * Get the organization a user belongs to
-     * @param uuid the user identifier
-     * @return a future organization
```
Comment (emmadeno) [inline app/src/main/java/com/ncnf/repositories/OrganizationRepository.java:69]: why did you remove all these methods ? Were we not using them ?

File: build.gradle [hunk 12]
```diff
@@ -5,7 +5,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath "com.android.tools.build:gradle:4.1.3"
+        classpath 'com.android.tools.build:gradle:4.1.2'
         classpath 'com.google.gms:google-services:4.3.5'
         classpath 'com.google.dagger:hilt-android-gradle-plugin:2.33-beta'
```
Comment (emmadeno) [inline build.gradle:11]: Should this be in the PR ? Especially since it's downgrading the version....

**Review by emmadeno — APPROVED** [review]
Looks good, you'll just have to remove the unnecessary files :) A lot of the code a lot cleaner and better tested now, great job !

File: app/src/main/java/com/ncnf/utilities/Helpers.java (hunk 1 above)
Comment (eloigrndl) [reply app/src/main/java/com/ncnf/utilities/Helpers.java:10]: I moved it here because this is just a generic operation performed on Future so it is not related to the DB.

File: app/src/main/java/com/ncnf/utilities/map/MapHandler.java (hunk 3 above)
Comment (eloigrndl) [reply app/src/main/java/com/ncnf/utilities/map/MapHandler.java:129]: Just to make sure that people understand what it does even if the method is private yes

File: app/src/main/java/com/ncnf/views/fragments/organization/OrganizationTabFragment.java (hunk 5 above)
Comment (eloigrndl) [reply app/src/main/java/com/ncnf/views/fragments/organization/OrganizationTabFragment.java:193]: I removed the view argument as it wasn't used in the method

File: .idea/compiler.xml (hunk 6 above)
Comment (eloigrndl) [reply .idea/compiler.xml:6]: yes you're right they should be there

File: app/src/main/java/com/ncnf/repositories/OrganizationRepository.java (hunk 11 above)
Comment (eloigrndl) [reply app/src/main/java/com/ncnf/repositories/OrganizationRepository.java:69]: I think when doing the automatic refactoring intellIJ suggested to erase it since it wasn't used so we might put it back later yes.

File: build.gradle (hunk 12 above)
Comment (eloigrndl) [reply build.gradle:11]: I might have made a mistake here you are right
